File: express-ts/src/services/asmaulhusna.service.ts
import data from "../configs/asmaulhusna.json";
import { AppDataSource } from "../configs/typeorm.config";
import { AsmaulHusna } from "../entities/asmaulhusna.entity";

export class AsmaulHusnaService {
  public async getAsmaulHusna(): Promise<AsmaulHusna[]> {
    const response = await AppDataSource.getRepository(AsmaulHusna).find();
    return response;
  }

  public async insertLoopAsmaulHusna(): Promise<void> {
    const asmaulhusna = data;

    for (let i = 0; i < asmaulhusna.length; i++) {
      await AppDataSource.getRepository(AsmaulHusna)
        .create({
          arab: asmaulhusna[i].arab,
          arti: asmaulhusna[i].arti,
          latin: asmaulhusna[i].latin,
          urutan: asmaulhusna[i].urutan,
        })
        .save();
    }
  }
}

File: golang/internal/services/asmaulhusna_service.go
package services

import (
	"encoding/json"
	"io"
	"os"

	"github.com/haikelz/asmaul-husna-api/internal/configs"
	"github.com/haikelz/asmaul-husna-api/internal/entities"
)

func GetAsmaulHusna() ([]*entities.AsmaulHusna, error) {
	var asmaulHusna []*entities.AsmaulHusna

	if err := configs.Gorm().Find(&asmaulHusna).Error; err != nil {
		return nil, err
	}

	return asmaulHusna, nil
}

func InsertLoopAsmaulHusna() ([]entities.AsmaulHusna, error) {
	// search where the file
	jsonFile, err := os.Open("internal/configs/asmaulhusna.json")

	if err != nil {
		return nil, err
	}

	defer jsonFile.Close()

	// Read the JSON file and convert the data to array byte
	fileData, _ := io.ReadAll(jsonFile)

	var asmaulHusna []entities.AsmaulHusna

	errLoad := json.Unmarshal(fileData, &asmaulHusna)

	if errLoad != nil {
		return nil, errLoad
	}

		for i := range asmaulHusna {
			data := &entities.AsmaulHusna{Urutan: asmaulHusna[i].Urutan, Latin: asmaulHusna[i].Latin, Arab: asmaulHusna[i].Arab, Arti: asmaulHusna[i].Arti}

			if dbErr := configs.Gorm().Create(&data).Error; dbErr != nil {
			}
		}

	return asmaulHusna, nil
}
